src/unix/system.c

```diff
@@ -341,7 +345,27 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (!xdg_data_home_dir) {
+	    xdg_data_home_dir = malloc(sizeof(char) * 
+				       (strlen(homedir) + strlen("/.local/share") + 1));
+	    malloc_needed = 1;
+	    check_snprintf = snprintf(xdg_data_home_dir, 
+				      strlen(homedir) + strlen("/.local/share") + 1, 
+		     "%s/%s", homedir, ".local/share");
+            if (check_snprintf < 0) {
+		    Sys_Error("xdg_data_home_dir: snprintf failed.\n");
+	    }
```

Comment:
```suggestion
        if (check_snprintf < 0) 
		    Sys_Error("xdg_data_home_dir:`XDG_DATA_HOME` not set, and and storing the path failed.\n");
```

Although removing the curly braces is a matter of taste, the error message is non-negotiable. It needs to explain what went wrong to a user.

Reply:
This is a matter of taste, but not a necessary one I believe. The file is filled with mixed styles relating to this. At first, I thought it was curly braces if in a block (not including functions), but then there are examples of ``if`` statements with braces and without nesting. I don't think this is necessary.

Reply:
Improved error message in 7ae9ce53.